server/pkg/controller/user/userauth.go

```diff
@@ -362,9 +362,13 @@ func (c *UserController) onVerificationSuccess(context *gin.Context, email strin
 	userID, err := c.UserRepo.GetUserIDWithEmail(email)
 	if err != nil {
 		if errors.Is(err, sql.ErrNoRows) {
-			userID, _, err = c.createUser(email, source)
-			if err != nil {
-				return ente.EmailAuthorizationResponse{}, stacktrace.Propagate(err, "")
+			if !viper.GetBool("internal.disable-registration") {
```

Comment:
Let us swap the conditions to avoid the double negation
```
if viper.GetBool("internal.disable-registration") {
 ... err
} else {
  ... normal
```